### DISCUSSION TITLE
Does the Flet framework utilize Flutter for web apps or implement its own solution for better SEO support?

### DISCUSSION THREAD

**Discussion by raslanismail87** [flet-dev/flet#1311] [Q&A] [answered]
### Question

I am aware that Flutter for web has limited SEO support. I was wondering if the Flet framework relies on Flutter for its web app development or if it has its own implementation to address the SEO limitations.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by kalexnorver** [comment, marked as answer]
The Flet framework is an open-source framework that aims to simplify and streamline the process of building web and mobile applications with Flutter. While Flutter for web has limited SEO support, the Flet framework does not rely solely on Flutter for its web app development. Instead, it implements its own solution for better SEO support.

The Flet framework utilizes server-side rendering (SSR) to create web pages that can be easily crawled and indexed by search engines. This approach allows the framework to generate static HTML pages for each route, which can then be served to clients instead of relying solely on JavaScript to render the page. By doing so, the Flet framework can improve the SEO of web applications built with Flutter.

As an example, consider a website that provides information about the popular game [PUBG supported symbols](https://elitepbgname.com/cool-symbols-for-pubg/). If this website is built with the Flet framework, it can generate static HTML pages for each section of the website, such as the homepage, the news section, and the forums. This allows search engines to crawl and index the website's content, making it easier for users to find the information they need. Additionally, the Flet framework's SSR approach can help improve the website's loading speed, which can also impact its SEO ranking.

**Reply by raslanismail87** [reply]
That is amazing, thank you

**Reply by ndonkoHenri** [reply]
Haha, I don't know why, but while reading I had the impression 'this answer was given by some AI chatbot :)'

**Reply by Xynonners** [reply]
does SSR only work for the HTML renderer?

for some reason even when I set web_renderer=ft.WebRenderer.HTML inspect element still tells me CanvasKit (requested explicitly)

**Reply by efrd526** [reply]
I faced the same SEO issue with Flutter Web and found that prerendering can be an effective solution.

Since Flutter renders via canvas, crawlers can’t index content properly. To fix this, I used Cloudflare Workers + prerender.io to serve static HTML to bots and the standard SPA to users.

After setup, my pages started indexing, rich snippets appeared, and I reached top 10 results in a competitive niche — all without switching from Flutter.

Happy to share more details if needed.

**Reply by Xynonners** [reply]
> I faced the same SEO issue with Flutter Web and found that prerendering can be an effective solution.
> 
> Since Flutter renders via canvas, crawlers can’t index content properly. To fix this, I used Cloudflare Workers + prerender.io to serve static HTML to bots and the standard SPA to users.
> 
> After setup, my pages started indexing, rich snippets appeared, and I reached top 10 results in a competitive niche — all without switching from Flutter.
> 
> Happy to share more details if needed.

Very cool approach. But how do you automate it?
